File: problems-extend-adapt/2601-2700/2678_passengers-over-sixty/solution.go
// The age is the two-digit field at offsets 11-12; char-code arithmetic
// decodes it without building a substring. The count is at most
// len(details) <= 100, so int is plenty.
func countOverSixty(details []string) int {
	count := 0
	for _, record := range details {
		age := int(record[11]-'0')*10 + int(record[12]-'0')
		if age > 60 {
			count++
		}
	}
	return count
}

File: problems-extend-adapt/2601-2700/2678_passengers-over-sixty/solution.rs
impl Solution {
    pub fn count_over_sixty(details: Vec<String>) -> i32 {
        // The age is the two-digit field at offsets 11-12; char-code
        // arithmetic decodes it without building a substring. The count is
        // at most details.len() <= 100, so i32 is plenty.
        let mut count = 0;
        for record in &details {
            let digits = record.as_bytes();
            let age = (digits[11] - b'0') as i32 * 10 + (digits[12] - b'0') as i32;
            if age > 60 {
                count += 1;
            }
        }
        count
    }
}
